pennylane/math/decomposition.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains methods that decomposes unitary matrices."""
+
+import numpy as np
+
+from pennylane import math
+
+
+def zyz_rotation_angles(U, return_global_phase=False):
+    r"""Compute the rotation angles :math:`\phi`, :math:`\theta`, and :math:`\omega` and the
+    phase :math:`\alpha` of a 2x2 unitary matrix as a product of Z and Y rotations in the form
+    :math:`e^{i\alpha} RZ(\omega) RY(\theta) RZ(\phi)`
+
+    Args:
+        U (array): 2x2 unitary matrix
+        return_global_phase (bool): if True, returns the global phase as well.
+
+    Returns:
+        tuple: The rotation angles :math:`\phi`, :math:`\theta`, and :math:`\omega` and the
+            global phase :math:`\alpha` if ``return_global_phase=True``.
+
+    """
+
+    U, alpha = math.convert_to_su2(U, return_global_phase=True)
+
+    abs_b = math.clip(math.abs(U[..., 0, 1]), 0, 1)
+    theta = 2 * math.arcsin(abs_b)
+
+    EPS = math.finfo(U.dtype).eps
+    half_phi_plus_omega = math.angle(U[..., 1, 1] + EPS)
```

Comment:
It used to be `atan2` which has a range `[-pi, pi]` but `angle` ranges within `(-pi, pi]`. Would this be the reason why there used to be interesting inconsistencies between test results?

Reply:
That was my guess at least. I thought using `qml.angle` would lead to more consistent results.

Reply:
Yeah anyway our previous 2qb decomp code doesn't read like any golden standard to me let's move on with `angle`